src/modules/dataFetching/hooks/useUser.tsx

```diff
@@ -0,0 +1,23 @@
+import apolloClient from '@/providers/apolloClient';
+import { UserFieldsFragmentDoc, useGetUserQuery } from '@/types/gqlTypes';
+
+export function useUser(userId?: string) {
```

Comment:
Its not slow, but what happens without it is:
1. Load table of all users. At this point we have all the user's names.
2. Click on one row. The `UserAuditPage` renders. The page title has the users name in it, so it makes a fetch request for the name, and shows a loading spinner until it returns. (This is the step that's cut out by just checking the fragment).
3. `UserAuditPage` proceeds to load the `UserAccessHistoryTable`, which makes another request to fetch the access history for this user. User sees a loading spinner until it completes.

That flow just bothers me, because the user is faced with 2 loading states, and one of them is for data that they already have in the cache (user's name).

There are other ways it could be resolved, and maybe they are less complex.  We could block the whole page behind a loading spinner until everything is loaded, and batch the requests. Or we could show an intermediate page title like "User Access History" until the name is returned along with everything else.

I prefer the fragment approach just because the UI appears more responsive on navigation.